Improving spring boot docs (#2496)
getsentry/sentry-docs: src/platforms/java/guides/spring-boot/index.mdx

### Recording Custom User Information

> In order to record custom user information, you can register a bean that implements `SentryUserProvider` interface.

Reviewer: ```suggestion
To record custom user information, you can register a bean that implements `SentryUserProvider` interface.
```

### Registering Custom Before Send Callback

> A Spring bean implementing `BeforeSendCallback` will be automatically set on `SentryOptions` during Sentry SDK auto-configuration. Note that there can be only a single bean like that.

Reviewer: A bit confusing, this might be more clear:
```suggestion
A Spring bean implementing the `BeforeSendCallback` will be automatically set on `SentryOptions` during the SDK's auto-configuration. Note that there can be only a single bean set this way.
```

### Registering Custom Before Breadcrumb Callback

> A Spring bean implementing `BeforeBreadcrumbCallback` will be automatically set on `SentryOptions` during Sentry SDK auto-configuration. Note that there can be only a single bean like that.

Reviewer: see previous edit

> A Spring bean implementing `BeforeBreadcrumbCallback` will be automatically set on `SentryOptions` during Sentry SDK auto-configuration. Note that there can be only a single bean like that.

Author: ```suggestion
A Spring bean implementing the `BeforeBreadcrumbCallback` will be automatically set on `SentryOptions` during the SDK's auto-configuration. Note that there can be only a single bean set this way.
```
